File: implementation/scipamato/core/sync/src/main/java/ch/difty/scipamato/core/sync/jobs/paper/SyncShortFieldConcatenator.java
package ch.difty.scipamato.core.sync.jobs.paper;

import java.sql.ResultSet;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * Gathers the content for the the fields methods, population and result based on a
 * resultSet of papers.
 * <p>
 * The content may either be those fields themselves or some concatenated short fields.
 */
interface SyncShortFieldConcatenator {

    /**
     * @param rs
     *     the paper recordset
     * @return the methods field content
     */
    @Nullable
    String methodsFrom(@NotNull ResultSet rs);

    /**
     * @param rs
     *     the paper recordset
     * @return the population field content
     */
    @Nullable
    String populationFrom(@NotNull ResultSet rs);

    /**
     * @param rs
     *     the paper recordset
     * @return the result field content
     */
    @Nullable
    String resultFrom(@NotNull ResultSet rs);
}

File: core/core-sync/src/main/kotlin/ch/difty/scipamato/core/sync/jobs/paper/SyncShortFieldConcatenator.kt
package ch.difty.scipamato.core.sync.jobs.paper

import java.sql.ResultSet

/**
 * Gathers the content for the fields `methods`, `population` and
 * `result` based on a resultSet of papers. The content may either
 * be those fields themselves or some concatenated short fields.
 */
interface SyncShortFieldConcatenator {
    fun methodsFrom(rs: ResultSet): String?
    fun populationFrom(rs: ResultSet): String?
    fun resultFrom(rs: ResultSet): String?
}
